### ISSUE TITLE
$compile('<my-component/>') doesn't work

### ISSUE THREAD

**Issue by pcable** [ngVue/ngVue#84]
If my markup contains `<my-component/>` the component will be rendered correctly. However if I call the angular $compile method 

`$compile('<my-component/>')($rootScope).html()` I get an empty string.

After considerable frustration I found that wrapping it with a div works. 

Raising this for the attention of others who might be so unfortunate.

**Comment by nicolaspayot** [comment]
I don't understand how can `<my-component />` be rendered correctly as self-closing tags for custom elements don't work with AngularJS. Here is a quote from Igor Minar: 

> self-closing or void elements as the html spec defines them are very special to the browser parser. you can't make your own, so for your custom elements you have to stick to non-void elements (<foo></foo>).
> 
> this can't be changed in angular. closing.

https://github.com/angular/angular.js/issues/1953#issuecomment-13135021

I think it should work with `$compile('<my-component></my-component>')($rootScope).html()`

**Comment by pcable** [comment]
Hi @nicolaspayot, Thanks for looking into this.

I guess it must be the browser that's permitting the use of the self closing tag (Chrome 72.0.3626). 

> I think it should work with $compile('<my-component></my-component>')($rootScope).html()

I've written some karma tests to show that it doesn't work without having a wrapper div regardless of self closing tags or not. This is different to the angular behavior which doesn't require the outer div

```javascript
angular
    .module('test', ['ngVue'])
    .component('myAng', {
        template: "<span>Hello From Angular</span>"
    })
    .value('myVue', {
        template: "<span>Hello From Vue</span>"
    })
    .directive('myVue', function (createVueComponent) {
        return createVueComponent('myVue')
    })

describe('ngVue compilation', () => {

    var $compile;
    var $rootScope;
    
    beforeEach(() => {
        angular.mock.module('test');
        angular.mock.inject((_$compile_, _$rootScope_) => {
            $compile = _$compile_
            $rootScope = _$rootScope_
        })
    })

    it('Renders angular components with no wrapper', () => {
        var contentShort = $compile('<my-ang/>')($rootScope.$new());
        expect(contentShort.text()).toBe("Hello From Angular")

        var contentLong = $compile('<my-ang></my-ang>')($rootScope.$new());
        expect(contentLong.html()).toBe(contentShort.html())
    })
    
    it('Renders vue components with wrapper', () => {
        var contentShort = $compile('<div><my-vue/></div>')($rootScope.$new());
        expect(contentShort.text()).toBe("Hello From Vue")

        var contentLong = $compile('<div><my-vue></my-vue></div>')($rootScope.$new());
        expect(contentLong.html()).toBe(contentShort.html())
    })

    //THIS ONE FAILS!
    it('Renders vue components with no wrapper', () => {
        var contentLong = $compile('<my-vue></my-vue>')($rootScope.$new());
        expect(contentLong.text()).toBe("Hello From Vue")
    })

})
```
This could cause an issue for anyone who is trying to do a direct conversion from their angular components into vue (which is where I had the issue) and this request should save them a lot of trouble.

**Comment by nicolaspayot** [comment]
Fixed with #85
